File: lab2/Classes/Center.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab2.Classes
{
    public class Center : Player
    {
        public int playerID { get; set; }
        public virtual Player player { get; set; }
        public double blocksPerGame { get; set; }
        public double reboundsPerGame { get; set; }
        public int blocks { get; set; } = 0;
        public int rebounds { get; set; } = 0;
        public Center(int id, string name, int height, int jerseyNumber, string type, double bpg, double rpg) : base(id, name, height, jerseyNumber, type)
        {
            blocksPerGame = bpg;
            reboundsPerGame = rpg;
            Console.WriteLine("Center is created");
        }

        public Center(int id, string name, int height, int jerseyNumber, string type, int blocks, int rebounds, double bpg, double rpg) : base(id, name, height, jerseyNumber, type) 
        {
            blocksPerGame = bpg;
            reboundsPerGame = rpg;
            this.blocks = blocks;
            Console.WriteLine("Center is created with blocks");
        }

        public void block() 
        {
            blocks++;
            Console.WriteLine($"blocks: {blocks}");
        }
        public void rebound() 
        {
            rebounds++;
            Console.WriteLine($"rebounds: {rebounds}");
        }

        public void setScreen() 
        {
            Console.WriteLine($"{Name} sets a screen for a teammate.");
        }

        public void post() 
        {
            Console.WriteLine($"{Name} is posting up in the paint.");
        }
        public override void play()
        {
            Console.WriteLine($"{Name} dominates the post, blocks and dunks the ball.");
        }
        public override void train()
        {
            Console.WriteLine($"{Name} is training playing close to basket, rebounding and blocking shots.");
        }
        public override void printInfo()
        {
            base.printInfo();
            Console.WriteLine($"Blocks per game : {blocksPerGame}");
            Console.WriteLine($"Rebounds per game : {reboundsPerGame}");
        }
    }
}

File: lab4/backend/database/models/Center.py
from typing import List, Optional, TYPE_CHECKING
from sqlalchemy import ForeignKey
from sqlalchemy import String
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column 
from sqlalchemy.orm import relationship
from ..base import Base

if TYPE_CHECKING:
    from .player import Player
class Center(Base):
    __tablename__ = "centers"

    id: Mapped[int] = mapped_column(ForeignKey("players.id"), primary_key=True)
    blocks: Mapped[int] = mapped_column(nullable=True)
    rebounds: Mapped[int] = mapped_column(nullable=True)
    blocks_per_game: Mapped[float] = mapped_column(nullable=True)
    rebounds_per_game: Mapped[float] = mapped_column(nullable=True)

    player: Mapped["Player"] = relationship(back_populates="center")
    
    def play(self):
        return str.format("{name} dominates the post, blocks and dunks the ball.", name=self.getPlayer().getName())
    
    def train(self):
        return str.format("{name} is training playing close to basket, rebounding and blocking shots.", name=self.getPlayer().getName())
    
    def printInfo(self):
        return self.getPlayer().getBasicInfo() + f"\nBlocks: {self.blocks}\nRebounds: {self.rebounds}\nBlocks per game: {self.blocks_per_game}\nRebounds per game: {self.rebounds_per_game}"
        
    def block(self):
        self.blocks += 1
        return str.format("blocks: {blocks}", blocks=self.blocks)
    
    def rebound(self):
        self.rebounds += 1
        return str.format("rebounds: {rebounds}", rebounds=self.rebounds)
    
    def setScreen(self):
        return str.format("{name} sets a screen.", name=self.getPlayer().getName())
    
    def post(self):
        return str.format("{name} is posting up in the paint.", name=self.getPlayer().getName())

    def __repr__(self) -> str:
        return f"Center(id={self.id!r}, blocks={self.blocks!r}, rebounds={self.rebounds!r}, blocks_per_game={self.blocks_per_game!r}, rebounds_per_game={self.rebounds_per_game!r})"
    
    def getPlayer(self):
        return self.player